libnetwork/controller.go

```diff
@@ -922,6 +923,11 @@ func (c *Controller) NewSandbox(containerID string, options ...SandboxOption) (_
 
 	sb.processOptions(options...)
 
+	// Set the initial value of ipv6Disabled based on whether the host has IPv6. This
+	// value will be used for host networking. For other types of networking, it will
+	// be updated once the container has been created and can be inspected.
+	sb.ipv6Disabled = !netutils.IsV6Listenable()
```

Comment:
I foresee the potential for lots of subtle bugs by having `sb.ipv6Disabled` exist. For example, if `NewSandbox` is called with the ID of a container that already has a sandbox  in `c.sandboxes`, `sb.ipv6Disabled` would get unconditionally cleared even if `sb.osSbox.IPV6LoEnabled() == false`. Or I can imagine some future contributor coming in, assuming incorrectly that `ipv6Disabled` _controls_ disabling IPv6 in a sandbox, and submitting some misguided PR that forces `sb.ipv6Disabled = true`.

We have the sources of truth readily available: `netutils.IsV6Listenable()` and `sb.osSbox.IPv6LoEnabled()`. Why do we need to cache it anywhere? If we do need to cache it for performance or other reasons, why do it in the ostensibly OS-independent `Sandbox` struct instead of the `osl.Namespace`?

Reply:
I've added a `Sandbox.ipv6Enabled()` and, because it involves a netlink call and will be needed in a few couple places by follow-up changes, cached the result in `osl.Sandbox`.